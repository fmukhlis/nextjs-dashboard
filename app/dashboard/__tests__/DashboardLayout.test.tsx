import { render, screen } from "@testing-library/react";
import DashboarLayout from "../layout";

const DummyChild = () => <div>Dummy Child</div>;

describe("DashboardLayout", () => {
  it("renders the SideNav", () => {
    render(
      <DashboarLayout>
        <DummyChild />
      </DashboarLayout>,
    );
    const acmeLogo = screen.getByText(/Acme/i);
    expect(acmeLogo).toBeInTheDocument();
  });

  it("renders children properly", () => {
    render(
      <DashboarLayout>
        <DummyChild />
      </DashboarLayout>,
    );
    const dummyChild = screen.getByText(/Dummy Child/i);
    expect(dummyChild).toBeInTheDocument();
  });

  // Snapshot
  it("renders DashboardLayout unchanged", () => {
    const { container } = render(
      <DashboarLayout>
        <DummyChild />
      </DashboarLayout>,
    );
    expect(container).toMatchSnapshot();
  });
});
