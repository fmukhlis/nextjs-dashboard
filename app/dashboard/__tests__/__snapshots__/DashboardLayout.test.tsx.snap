// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DashboardLayout renders DashboardLayout unchanged 1`] = `
<div>
  <div
    class="flex h-screen flex-col md:flex-row md:overflow-hidden"
  >
    <div
      class="w-full flex-none md:w-64"
    >
      <div
        class="flex h-full flex-col px-3 py-4 md:px-2"
      >
        <a
          class="mb-2 flex h-20 items-end justify-start rounded-md bg-blue-600 p-4 md:h-40"
          href="/"
        >
          <div
            class="w-44 text-white md:w-52"
          >
            <div
              class="className flex flex-row items-center leading-none text-white"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-earth h-12 w-12 rotate-45"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.54 15H17a2 2 0 0 0-2 2v4.54"
                />
                <path
                  d="M7 3.34V5a3 3 0 0 0 3 3a2 2 0 0 1 2 2c0 1.1.9 2 2 2a2 2 0 0 0 2-2c0-1.1.9-2 2-2h3.17"
                />
                <path
                  d="M11 21.95V18a2 2 0 0 0-2-2a2 2 0 0 1-2-2v-1a2 2 0 0 0-2-2H2.05"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
              </svg>
              <p
                class="text-[44px] tracking-wider ml-3"
              >
                Acme
              </p>
            </div>
          </div>
        </a>
        <div
          class="flex grow flex-row justify-between space-x-2 md:flex-col md:space-x-0 md:space-y-2"
        >
          <a
            class="flex h-[48px] grow items-center justify-center gap-2 rounded-md bg-gray-50 p-3 text-sm font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
            href="/dashboard"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-house w-6"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
              />
              <path
                d="M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
              />
            </svg>
            <p
              class="hidden md:block"
            >
              Home
            </p>
          </a>
          <a
            class="flex h-[48px] grow items-center justify-center gap-2 rounded-md bg-gray-50 p-3 text-sm font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
            href="/dashboard/invoices"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-files w-6"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 2a2 2 0 0 1 1.414.586l4 4A2 2 0 0 1 21 8v7a2 2 0 0 1-2 2h-8a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2z"
              />
              <path
                d="M15 2v4a2 2 0 0 0 2 2h4"
              />
              <path
                d="M5 7a2 2 0 0 0-2 2v11a2 2 0 0 0 2 2h8a2 2 0 0 0 1.732-1"
              />
            </svg>
            <p
              class="hidden md:block"
            >
              Invoices
            </p>
          </a>
          <a
            class="flex h-[48px] grow items-center justify-center gap-2 rounded-md bg-gray-50 p-3 text-sm font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
            href="/dashboard/customers"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-users w-6"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
              />
              <path
                d="M16 3.128a4 4 0 0 1 0 7.744"
              />
              <path
                d="M22 21v-2a4 4 0 0 0-3-3.87"
              />
              <circle
                cx="9"
                cy="7"
                r="4"
              />
            </svg>
            <p
              class="hidden md:block"
            >
              Customers
            </p>
          </a>
          <div
            class="hidden h-auto w-full grow rounded-md bg-gray-50 md:block"
          />
          <form>
            <button
              class="flex h-[48px] w-full grow items-center justify-center gap-2 rounded-md bg-gray-50 p-3 text-sm font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-power w-6"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2v10"
                />
                <path
                  d="M18.4 6.6a9 9 0 1 1-12.77.04"
                />
              </svg>
              <div
                class="hidden md:block"
              >
                Sign Out
              </div>
            </button>
          </form>
        </div>
      </div>
    </div>
    <div
      class="flex-grow p-6 md:overflow-y-auto md:p-12"
    >
      <div>
        Dummy Child
      </div>
    </div>
  </div>
</div>
`;
